internal/ui/keyboard_js.go

```diff
@@ -0,0 +1,39 @@
+// +build js
+
+package ui
+
+import "github.com/gopherjs/gopherjs/js"
+
+var doc = js.Global.Get("document")
+
+func push(char rune) {
+	go func() {
+		rblock.Lock()
+		runebuffer = append(runebuffer, char)
+		rblock.Unlock()
+	}()
+}
+
+func init() {
+	if doc == nil {
+		return
+	}
+	kp := doc.Get("onkeypress")
```

Comment:
All have keypress, the weird thing is charCode is deprecated in favor of the unimplemented char